lib/broadway_dashboard.ex

```diff
@@ -19,31 +19,51 @@ defmodule BroadwayDashboard do
 
   @impl true
   def init(opts) do
-    pipelines = Keyword.get(opts, :pipelines, [])
+    pipelines = opts[:pipelines] || :auto_discover
 
     {:ok, %{pipelines: pipelines}, application: :broadway}
   end
 
   @impl true
   def menu_link(%{pipelines: pipelines}, _capabilities) do
-    case pipelines do
-      [] ->
-        {:disabled, @page_title, @disabled_link}
+    if pipelines == [] do
+      {:disabled, @page_title, @disabled_link}
+    else
+      {:ok, @page_title}
+    end
+  end
 
-      [_ | _] ->
-        {:ok, @page_title}
+  defp pipelines_or_auto_discover(pipeline_config, node) do
+    if pipeline_config == :auto_discover do
+      with {:broadway, :ok} <- {:broadway, check_broadway_version(node)},
+           {:badrpc, _error} <- :rpc.call(node, Broadway, :all_running, []) do
+        {{:error, :cannot_list_running_pipelines}, []}
+      else
+        pipelines when is_list(pipelines) ->
+          {:ok, pipelines}
```

Comment:
Maybe you should convert the empty list case to an error here? 🤔

Reply:
Agree!